File: EndGate/EndGate.Core.Client.JS/Utilities/LooperCallback.js
var EndGate;
(function (EndGate) {
    (function (Core) {
        (function (Utilities) {
            var LooperCallback = (function () {
                function LooperCallback(fps, callback) {
                    this._type = "LooperCallback";
                    this.Fps = fps;
                    this.Callback = callback;
                    this.TimeoutID = 0;
                    this.ID = LooperCallback._ids++;
                }
                LooperCallback._ids = 0;
                return LooperCallback;
            })();
            Utilities.LooperCallback = LooperCallback;            
        })(Core.Utilities || (Core.Utilities = {}));
        var Utilities = Core.Utilities;
    })(EndGate.Core || (EndGate.Core = {}));
    var Core = EndGate.Core;
})(EndGate || (EndGate = {}));

File: EndGate/EndGate.Core/Utilities/LooperCallback.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EndGate.Core.Utilities
{
    internal class LooperCallback
    {
        private int _fps;

        public LooperCallback()
        {
        }

        public LooperCallback(int fps, Action callback)
        {
            Fps = fps;
            Callback = callback;
        }

        public int TriggerFrequency { get; private set; }
        public long LastTriggered { get; set; }
        public Action Callback { get; set; }
        public int Fps
        {
            get
            {
                return _fps;
            }
            set
            {
                _fps = value;

                TriggerFrequency = (_fps != 0) ? 1000 / _fps : 0;
            }
        }
    }
}
